ffi/derive/src/ffi_fn.rs

```diff
@@ -61,21 +49,42 @@ pub fn gen_definition(fn_descriptor: &FnDescriptor) -> TokenStream {
     }
 }
 
-pub fn gen_fn_name(self_ty_name: Option<&Ident>, method_name: &Ident) -> Ident {
-    let self_ty_name = self_ty_name.map_or_else(Default::default, ToString::to_string);
+pub fn gen_fn_name(fn_descriptor: &FnDescriptor) -> Ident {
+    let method_name = &fn_descriptor.sig.ident;
+    let self_ty_name = fn_descriptor
+        .self_ty_name()
+        .map_or_else(Default::default, ToString::to_string);
+    let trait_name = fn_descriptor
+        .trait_name()
+        .map_or_else(Default::default, |trait_name| format!("__{trait_name}"));
 
     Ident::new(
-        &format!("{}__{}", self_ty_name, method_name),
+        &format!("{}{}__{}", self_ty_name, trait_name, method_name),
         proc_macro2::Span::call_site(),
     )
 }
 
-fn gen_doc(self_type: Option<&Ident>, method_name: &Ident) -> String {
+fn gen_doc(fn_descriptor: &FnDescriptor) -> String {
     // NOTE: [#docs = "some_doc"] expands to ///some_doc, therefore the leading space
+    let method_name = &fn_descriptor.sig.ident;
+    let self_type = fn_descriptor
+        .self_ty
+        .as_ref()
+        .and_then(syn::Path::get_ident);
+    let trait_name = fn_descriptor
+        .trait_name
+        .as_ref()
+        .and_then(syn::Path::get_ident);
 
     let path = self_type.map_or_else(
         || method_name.to_string(),
-        |self_ty| format!("{}::{}", self_ty, method_name),
+        |self_ty| {
+            trait_name.map_or_else(
+                || format!("{}::{}", self_ty, method_name),
+                // NOTE: Fully-qualified syntax currently not supported
```

Comment:
you mean like `<Self as Trait>::method_name`?. I think we don't have to support that for the docs

Reply:
Yeah, there is [issue](https://github.com/rust-lang/rust/issues/74563) for that.
For now i stick with link to `Trait::method_name`.

Reply:
Low priority.